src/vsg/platform/win32/Win32_Window.cpp

```diff
@@ -568,8 +568,8 @@ LRESULT Win32_Window::handleWin32Messages(UINT msg, WPARAM wParam, LPARAM lParam
         uint32_t mx = GET_X_LPARAM(lParam);
         uint32_t my = GET_Y_LPARAM(lParam);
 
-        uint32_t button = msg == WM_LBUTTONDOWN ? 1 : (msg == WM_RBUTTONDOWN ? 2 : msg == WM_MBUTTONDOWN ? 3 : (msg == WM_XBUTTONDOWN ? 4 : 0)); // need to determine x1, x2
-        _bufferedEvents.emplace_back(new vsg::ButtonReleaseEvent(this, event_time, mx, my, getButtonMask(wParam), getButtonEventDetail(msg)));
+        uint32_t button = msg == WM_LBUTTONUP ? 1 : (msg == WM_RBUTTONUP ? 2 : msg == WM_MBUTTONUP ? 3 : (msg == WM_XBUTTONUP ? 4 : 0)); // need to determine x1, x2
+        _bufferedEvents.emplace_back(new vsg::ButtonReleaseEvent(this, event_time, mx, my, getButtonMask(wParam), button));
```

Comment:
There may be a typo.Please check it.
@robertosfield

Reply:
A typo in your revision or @tomhog's original work on Win32_Window.cpp?

Reply:
A typo in vsg master branch.I think the 'msg' should be 'WM_XXXUP' rather than 'WM_XXXDOWN' for button release event.I fixed it and tested in my revision.

Reply:
Just had a look at the changes code.  The button value isn't used anywhere, so while the code looks wrong, it's also not used.  I'll just delete the line and check this into master.

Reply:
So I've remove the offending line and the automated build just went through under Windows so no problems.  My guess is that Tom originally put the line in as a copy and paste from other button handling and as the implementation evolved removed the original place as part of the clean up process but missed this one.

https://github.com/vsg-dev/VulkanSceneGraph/commit/c86b88f2e3b2ddb8794b65c554460b005fb528a9